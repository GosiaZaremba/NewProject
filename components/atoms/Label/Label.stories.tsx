import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { Label } from './Label';

storiesOf('atoms/Label', module)
    .addDecorator((story) => story())
    .add('default', () => <Label text={'Label'} />);
