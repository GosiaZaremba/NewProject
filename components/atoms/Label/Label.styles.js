import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  labelText: {
    fontSize: 16,
  },
});
