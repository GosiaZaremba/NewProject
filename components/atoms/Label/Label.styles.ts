import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  labelText: {
    fontSize: 16,
    textAlign: 'center',
  },
});
