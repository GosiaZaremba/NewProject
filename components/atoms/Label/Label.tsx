import React from 'react';
import { Text } from 'react-native';
import styles from './Label.styles';

export type Props = {
    text: string;
    testID?: string;
};

export const Label: React.FC<Props> = ({ text, testID }) => {
    return (
        <Text style={styles.text} testID={testID}>
            {text}
        </Text>
    );
};
