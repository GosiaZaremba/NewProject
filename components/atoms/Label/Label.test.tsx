import { render } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { Label } from './Label';

describe('Atom Label', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(<Label text={'Label text'} testID={'atom-label'} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure variants works correctly', () => {
        const { getByTestId } = render(
            <Label text={'Label text'} testID={'atom-label'} />
        );

        const label = getByTestId('atom-label');
        expect(label).toHaveTextContent('Label text');
    });
});
