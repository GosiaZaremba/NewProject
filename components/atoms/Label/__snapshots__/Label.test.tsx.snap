// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom Label Component Snapshot 1`] = `
<Text
  style={
    Object {
      "color": "black",
      "marginBottom": 2,
    }
  }
  testID="atom-label"
>
  Label text
</Text>
`;
